NPD in NewFrameSession

### Brief summary

NPD when calling `s.ID()` on an instance of session that was passed into `NewFrameSession`. The NPD occurs in the `ID` method itself, which is a simple `return s.id`, which means that the session instance is incomplete/nil (suggesting that since in the stach trace we have `(*Session).ID(0xc00334d8e8?)` and the `?` denotes _"uncertainty about the validity of the memory address or a possible truncation or mismatch of debug information"_)?

```bash
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x48 pc=0x15e6940]

goroutine 683891 [running]:
github.com/grafana/xk6-browser/common.(*Session).ID(0xc00334d8e8?)
github.com/grafana/xk6-browser@v1.9.2-0.20241118153201-d741e4540a3a/common/session.go:53
github.com/grafana/xk6-browser/common.NewFrameSession({0x1f9ad08, 0xc0027e8b90}, {0x1fa4cc0, 0x0}, 0xc000b11860, 0xc001b77ec0, {0xc0012c7840, 0x20}, 0xc002ba7920, 0x0)
github.com/grafana/xk6-browser@v1.9.2-0.20241118153201-d741e4540a3a/common/frame_session.go:95 +0x6e
github.com/grafana/xk6-browser/common.(*FrameSession).attachIFrameToTarget(0xc001b77ec0, 0xc000658bd0, {0xc0012c7820, 0x20})
github.com/grafana/xk6-browser@v1.9.2-0.20241118153201-d741e4540a3a/common/frame_session.go:1034 +0xe5
github.com/grafana/xk6-browser/common.(*FrameSession).onAttachedToTarget(0xc001b77ec0, 0xc003e81460)
github.com/grafana/xk6-browser@v1.9.2-0.20241118153201-d741e4540a3a/common/frame_session.go:967 +0x277
github.com/grafana/xk6-browser/common.(*FrameSession).initEvents.func1()
github.com/grafana/xk6-browser@v1.9.2-0.20241118153201-d741e4540a3a/common/frame_session.go:303 +0x233
created by github.com/grafana/xk6-browser/common.(*FrameSession).initEvents in goroutine 683768
github.com/grafana/xk6-browser@v1.9.2-0.20241118153201-d741e4540a3a/common/frame_session.go:239 +0x19b
```

Test run id: 3536152

### xk6-browser version

v1.9.1

### OS

NA

### Chrome version

NA

### Docker version and image (if applicable)

_No response_

### Steps to reproduce the problem

-

### Expected behaviour

-

### Actual behaviour

-

--- a/internal/js/modules/k6/browser/common/frame_session.go
+++ b/internal/js/modules/k6/browser/common/frame_session.go
@@ -948,15 +948,14 @@ func (fs *FrameSession) onAttachedToTarget(event *target.EventAttachedToTarget) 
 
 	switch ti.Type {
 	case "iframe":
-		err = fs.attachIFrameToTarget(ti, sid)
+		err = fs.attachIFrameToTarget(ti, session)
 	case "worker":
-		err = fs.attachWorkerToTarget(ti, sid)
+		err = fs.attachWorkerToTarget(ti, session)
 	default:
 		// Just unblock (debugger continue) these targets and detach from them.
-		s := fs.page.browserCtx.getSession(sid)
-		_ = s.ExecuteWithoutExpectationOnReply(fs.ctx, cdpruntime.CommandRunIfWaitingForDebugger, nil, nil)
-		_ = s.ExecuteWithoutExpectationOnReply(fs.ctx, target.CommandDetachFromTarget,
-			&target.DetachFromTargetParams{SessionID: s.id}, nil)
+		_ = session.ExecuteWithoutExpectationOnReply(fs.ctx, cdpruntime.CommandRunIfWaitingForDebugger, nil, nil)
+		_ = session.ExecuteWithoutExpectationOnReply(fs.ctx, target.CommandDetachFromTarget,
+			&target.DetachFromTargetParams{SessionID: session.id}, nil)
 	}
 	if err == nil {
 		return
@@ -1001,7 +1000,8 @@ func (fs *FrameSession) onAttachedToTarget(event *target.EventAttachedToTarget) 
 }
 
 // attachIFrameToTarget attaches an IFrame target to a given session.
-func (fs *FrameSession) attachIFrameToTarget(ti *target.Info, sid target.SessionID) error {
+func (fs *FrameSession) attachIFrameToTarget(ti *target.Info, session *Session) error {
+	sid := session.ID()
 	fr, ok := fs.manager.getFrameByID(cdp.FrameID(ti.TargetID))
 	if !ok {
 		// IFrame should be attached to fs.page with EventFrameAttached
@@ -1022,7 +1022,7 @@ func (fs *FrameSession) attachIFrameToTarget(ti *target.Info, sid target.Session
 
 	nfs, err := NewFrameSession(
 		fs.ctx,
-		fs.page.browserCtx.getSession(sid),
+		session,
 		fs.page, fs, ti.TargetID,
 		fs.logger,
 		false)
@@ -1039,13 +1039,13 @@ func (fs *FrameSession) attachIFrameToTarget(ti *target.Info, sid target.Session
 }
 
 // attachWorkerToTarget attaches a Worker target to a given session.
-func (fs *FrameSession) attachWorkerToTarget(ti *target.Info, sid target.SessionID) error {
-	w, err := NewWorker(fs.ctx, fs.page.browserCtx.getSession(sid), ti.TargetID, ti.URL)
+func (fs *FrameSession) attachWorkerToTarget(ti *target.Info, session *Session) error {
+	w, err := NewWorker(fs.ctx, session, ti.TargetID, ti.URL)
 	if err != nil {
 		return fmt.Errorf("attaching worker target ID %v to session ID %v: %w",
-			ti.TargetID, sid, err)
+			ti.TargetID, session.ID(), err)
 	}
-	fs.page.workers[sid] = w
+	fs.page.workers[session.ID()] = w
 
 	return nil
 }

